---
title: "metagene_siRNA"
author: "Chenxin Li"
date: "Sep 6, 2019"
output:
  html_notebook:
    number_sections: yes
    toc: yes
    toc_float: yes
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r}
library(ggplot2)
library(emmeans)
library(tidyr)
library(dplyr)
library(readr)
library(readxl)
library(vegan)
library(stringr)
library(RColorBrewer)
library(svglite)
```

#load data
##upstream

```{r}
BS1_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/blender-sperm-1_S30_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

BS2_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/BlenderSperm2_S8_L004_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

BS4a_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/BlenderSperm4a_S173_L007_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

BS4b_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/Blendersperm4b_S176_L007_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

BS6a_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/Blendersperm6a_S174_L007_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

BS6b_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/Blendersperm6b_S177_L007_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

B15_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/Bulk-15_S26_L004_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

B20_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/Bulk-20_S27_L004_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

B25_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/Bulk-25_S28_L004_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

ddm1_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/ddm1ab.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

ddm1_con_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/ddm1ab_control.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

drm2_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/drm2-T-DNA.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

drm2_con_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/drm2-T-DNA_control.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

E9a_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/egg-9a_S25_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

E9b_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/egg-9b_S26_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

E9c_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/egg-9c_S27_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

E2_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/eggcell2_combined.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

E3_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/eggcell3_combined.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

E4_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/eggcell4_combined.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

OV1a_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/ovary-no-egg-1a_S31_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

OV1b_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/ovary-no-egg-1b_S32_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

OV2_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/Ovarynoegg2_S5_L004_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

SD4_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/seedling4_S3_L004_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

SP5b_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/sperm-5b_S28_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

SP5c_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/sperm-5c_S29_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

root_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5713891.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

em_dry_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR2544786.filtered.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

em_12_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR2544787.filtered.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

em_24_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR2544788.filtered.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")
```

```{r}
em_7DAF_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR771501.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

en_7DAF_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR771500.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")
```


```{r}
S3_1_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048325_9522_S3_1_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

S3_2_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048326_9522_S3_2_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

S3_3_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048327_9522_S3_3_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

S5_1_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048328_9522_S5_1_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

S5_2_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048329_9522_S5_2_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

S5_3_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048330_9522_S5_3_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

S7_1_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048331_9522_S7_1_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

S7_2_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048332_9522_S7_2_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

S7_3_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048333_9522_S7_3_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")
```

```{r}
an_pre_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049778.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

an_mei_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049779.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

an_mic_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049780.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

an_bi_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049781.1.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")
```

```{r}
ov_1st_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049786.1.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

ov_2nd_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049787.1.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

ov_3rd_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049788.1.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")

ov_4th_up <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049789.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_upstream.txt")
```


```{r}
B15_up <- B15_up %>% 
  mutate(sample_ID = "B15")

B20_up <- B20_up %>% 
  mutate(sample_ID = "B20")

B25_up <- B25_up %>% 
  mutate(sample_ID = "B25")

BS1_up <- BS1_up %>% 
  mutate(sample_ID = "BS1")

BS2_up <- BS2_up %>% 
  mutate(sample_ID = "BS2")

BS4a_up <- BS4a_up %>% 
  mutate(sample_ID = "BS4a")

BS4b_up <- BS4b_up %>% 
  mutate(sample_ID = "BS4b")

BS6a_up <- BS6a_up %>% 
  mutate(sample_ID = "BS6a")

BS6b_up <- BS6b_up %>% 
  mutate(sample_ID = "BS6b")

ddm1_up <- ddm1_up %>% 
  mutate(sample_ID = "ddm1")

ddm1_con_up <- ddm1_con_up %>% 
  mutate(sample_ID = "ddm1_con")

drm2_up <- drm2_up %>% 
  mutate(sample_ID = "drm2")

drm2_con_up <- drm2_con_up %>% 
  mutate(sample_ID = "drm2_con")

E2_up <- E2_up %>% 
  mutate(sample_ID = "E2")

E3_up <- E3_up %>% 
  mutate(sample_ID = "E3")

E4_up <- E4_up %>% 
  mutate(sample_ID = "E4")

E9a_up <- E9a_up %>% 
  mutate(sample_ID = "E9a")

E9b_up <- E9b_up %>% 
  mutate(sample_ID = "E9b")

E9c_up <- E9c_up %>% 
  mutate(sample_ID = "E9c")

OV1a_up <- OV1a_up %>% 
  mutate(sample_ID = "OV1b")

OV1b_up <- OV1b_up %>% 
  mutate(sample_ID = "OV1b")

OV2_up <- OV2_up %>% 
  mutate(sample_ID = "OV2")

root_up <- root_up %>% 
  mutate(sample_ID = "root")

SD4_up <- SD4_up %>% 
  mutate(sample_ID = "SD4")

SP5b_up <- SP5b_up %>% 
  mutate(sample_ID = "SP5b")

SP5c_up <- SP5c_up %>% 
  mutate(sample_ID = "SP5c")

S3_1_up <- S3_1_up %>% 
  mutate(sample_ID = "S3_1")

S3_2_up <- S3_2_up %>% 
  mutate(sample_ID = "S3_2")

S3_3_up <- S3_3_up %>% 
  mutate(sample_ID = "S3_3")

S5_1_up <- S5_1_up %>% 
  mutate(sample_ID = "S5_1")

S5_2_up <- S5_2_up %>% 
  mutate(sample_ID = "S5_2")

S5_3_up <- S5_3_up %>% 
  mutate(sample_ID = "S5_3")

S7_1_up <- S7_1_up %>% 
  mutate(sample_ID = "S7_1")

S7_2_up <- S7_2_up %>% 
  mutate(sample_ID = "S7_2")

S7_3_up <- S7_3_up %>% 
  mutate(sample_ID = "S7_3")

an_pre_up <- an_pre_up %>% 
  mutate(sample_ID = "an_pre")

an_mei_up <- an_mei_up %>% 
  mutate(sample_ID = "an_mei")

an_mic_up <- an_mic_up %>% 
  mutate(sample_ID = "an_mic")

an_bi_up <- an_bi_up %>% 
  mutate(sample_ID = "an_bi")

ov_1st_up <- ov_1st_up %>% 
  mutate(sample_ID = "ov_1st")

ov_2nd_up <- ov_2nd_up %>% 
  mutate(sample_ID = "ov_2nd")

ov_3rd_up <- ov_3rd_up %>% 
  mutate(sample_ID = "ov_3rd")

ov_4th_up <- ov_4th_up %>% 
  mutate(sample_ID = "ov_4th")

em_dry_up <- em_dry_up %>% 
  mutate(sample_ID = "em_dry")

em_12_up <- em_12_up %>%
  mutate(sample_ID = "em_12h")

em_24_up <- em_24_up %>% 
  mutate(sample_ID = "em_24h")

em_7DAF_up <- em_7DAF_up %>% 
  mutate(sample_ID = "em_7DAF")

en_7DAF_up <- en_7DAF_up %>% 
  mutate(sample_ID = "en_7DAF")

upstream <- rbind(BS1_up, BS2_up, BS4a_up, BS4b_up, BS6a_up, BS6b_up, 
                  B15_up, B20_up, B25_up, 
                  ddm1_up, ddm1_con_up, drm2_up, drm2_con_up, 
                  E2_up, E3_up, E4_up, E9a_up, E9b_up, E9c_up,
                  OV1a_up, OV1b_up, OV2_up, 
                  root_up, SD4_up, SP5b_up, SP5c_up, 
                  S3_1_up, S3_2_up, S3_3_up, 
                  S5_1_up, S5_2_up, S5_3_up,
                  S7_1_up, S7_2_up, S7_3_up, 
                  an_pre_up, an_mei_up, an_mic_up, an_bi_up, 
                  ov_1st_up, ov_2nd_up, ov_3rd_up, ov_4th_up,
                  em_dry_up, em_12_up,
                  em_24_up, em_7DAF_up, en_7DAF_up) %>%  
  mutate(location = "5' end")
```



##downstream
```{r}
BS1_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/blender-sperm-1_S30_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

BS2_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/BlenderSperm2_S8_L004_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

BS4a_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/BlenderSperm4a_S173_L007_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

BS4b_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/Blendersperm4b_S176_L007_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

BS6a_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/Blendersperm6a_S174_L007_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

BS6b_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/Blendersperm6b_S177_L007_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

B15_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/Bulk-15_S26_L004_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

B20_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/Bulk-20_S27_L004_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

B25_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/Bulk-25_S28_L004_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

ddm1_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/ddm1ab.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

ddm1_con_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/ddm1ab_control.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

drm2_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/drm2-T-DNA.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

drm2_con_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/drm2-T-DNA_control.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

E9a_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/egg-9a_S25_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

E9b_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/egg-9b_S26_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

E9c_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/egg-9c_S27_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

E2_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/eggcell2_combined.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

E3_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/eggcell3_combined.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

E4_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/eggcell4_combined.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

OV1a_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/ovary-no-egg-1a_S31_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

OV1b_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/ovary-no-egg-1b_S32_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

OV2_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/Ovarynoegg2_S5_L004_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

SD4_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/seedling4_S3_L004_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

SP5b_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/sperm-5b_S28_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

SP5c_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/sperm-5c_S29_L002_R1_001.fastq.gz.trimmed.dedup.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

root_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5713891.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

em_dry_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR2544786.filtered.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

em_12_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR2544787.filtered.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

em_24_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR2544788.filtered.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

em_7DAF_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR771501.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

en_7DAF_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR771500.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

S3_1_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048325_9522_S3_1_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

S3_2_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048326_9522_S3_2_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

S3_3_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048327_9522_S3_3_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

S5_1_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048328_9522_S5_1_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

S5_2_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048329_9522_S5_2_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

S5_3_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048330_9522_S5_3_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

S7_1_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048331_9522_S7_1_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

S7_2_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048332_9522_S7_2_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

S7_3_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/GSM2048333_9522_S7_3_filtered.fasta.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

an_pre_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049778.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

an_mei_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049779.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

an_mic_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049780.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

an_bi_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049781.1.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

ov_1st_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049786.1.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

ov_2nd_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049787.1.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

ov_3rd_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049788.1.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")

ov_4th_down <- read.delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/24nt_meta_all_genes/all_genes/SRR5049789.fastq.gz.trimmed.fastq.gz.sorted.bam_MSU7_filtered_genes.gff_downstream.txt")
```

```{r}
B15_down <- B15_down %>% 
  mutate(sample_ID = "B15")

B20_down <- B20_down %>% 
  mutate(sample_ID = "B20")

B25_down <- B25_down %>% 
  mutate(sample_ID = "B25")

BS1_down <- BS1_down %>% 
  mutate(sample_ID = "BS1")

BS2_down <- BS2_down %>% 
  mutate(sample_ID = "BS2")

BS4a_down <- BS4a_down %>% 
  mutate(sample_ID = "BS4a")

BS4b_down <- BS4b_down %>% 
  mutate(sample_ID = "BS4b")

BS6a_down <- BS6a_down %>% 
  mutate(sample_ID = "BS6a")

BS6b_down <- BS6b_down %>% 
  mutate(sample_ID = "BS6b")

ddm1_down <- ddm1_down %>% 
  mutate(sample_ID = "ddm1")

ddm1_con_down <- ddm1_con_down %>% 
  mutate(sample_ID = "ddm1_con")

drm2_down <- drm2_down %>% 
  mutate(sample_ID = "drm2")

drm2_con_down <- drm2_con_down %>% 
  mutate(sample_ID = "drm2_con")

E2_down <- E2_down %>% 
  mutate(sample_ID = "E2")

E3_down <- E3_down %>% 
  mutate(sample_ID = "E3")

E4_down <- E4_down %>% 
  mutate(sample_ID = "E4")

E9a_down <- E9a_down %>% 
  mutate(sample_ID = "E9a")

E9b_down <- E9b_down %>% 
  mutate(sample_ID = "E9b")

E9c_down <- E9c_down %>% 
  mutate(sample_ID = "E9c")

OV1a_down <- OV1a_down %>% 
  mutate(sample_ID = "OV1a")

OV1b_down <- OV1b_down %>% 
  mutate(sample_ID = "OV1b")

OV2_down <- OV2_down %>% 
  mutate(sample_ID = "OV2")

root_down <- root_down %>% 
  mutate(sample_ID = "root")

SD4_down <- SD4_down %>% 
  mutate(sample_ID = "SD4")

SP5b_down <- SP5b_down %>% 
  mutate(sample_ID = "SP5b")

SP5c_down <- SP5c_down %>% 
  mutate(sample_ID = "SP5c")

S3_1_down <- S3_1_down %>% 
  mutate(sample_ID = "S3_1")

S3_2_down <- S3_2_down %>% 
  mutate(sample_ID = "S3_2")

S3_3_down <- S3_3_down %>% 
  mutate(sample_ID = "S3_3")

S5_1_down <- S5_1_down %>% 
  mutate(sample_ID = "S5_1")

S5_2_down <- S5_2_down %>% 
  mutate(sample_ID = "S5_2")

S5_3_down <- S5_3_down %>% 
  mutate(sample_ID = "S5_3")

S7_1_down <- S7_1_down %>% 
  mutate(sample_ID = "S7_1")

S7_2_down <- S7_2_down %>% 
  mutate(sample_ID = "S7_2")

S7_3_down <- S7_3_down %>% 
  mutate(sample_ID = "S7_3")

an_pre_down <- an_pre_down %>% 
  mutate(sample_ID = "an_pre")

an_mei_down <- an_mei_down %>% 
  mutate(sample_ID = "an_mei")

an_mic_down <- an_mic_down %>% 
  mutate(sample_ID = "an_mic")

an_bi_down <- an_bi_down %>% 
  mutate(sample_ID = "an_bi")

ov_1st_down <- ov_1st_down %>% 
  mutate(sample_ID = "ov_1st")

ov_2nd_down <- ov_2nd_down %>% 
  mutate(sample_ID = "ov_2nd")

ov_3rd_down <- ov_3rd_down %>% 
  mutate(sample_ID = "ov_3rd")

ov_4th_down <- ov_4th_down %>% 
  mutate(sample_ID = "ov_4th")

em_dry_down <- em_dry_down %>% 
  mutate(sample_ID = "em_dry")

em_12_down <- em_12_down %>%
  mutate(sample_ID = "em_12h")

em_24_down <- em_24_down %>% 
  mutate(sample_ID = "em_24h")

em_7DAF_down <- em_7DAF_down %>% 
  mutate(sample_ID = "em_7DAF")

en_7DAF_down <- en_7DAF_down %>% 
  mutate(sample_ID = "en_7DAF")

downstream <- rbind(BS1_down, BS2_down, BS4a_down, BS4b_down, BS6a_down, BS6b_down, 
                  B15_down, B20_down, B25_down, 
                  ddm1_down, ddm1_con_down, drm2_down, drm2_con_down, 
                  E2_down, E3_down, E4_down, E9a_down, E9b_down, E9c_down,
                  OV1a_down, OV1b_down, OV2_down, 
                  root_down, SD4_down, SP5b_down, SP5c_down,  
                  S3_1_down, S3_2_down, S3_3_down, 
                  S5_1_down, S5_2_down, S5_3_down,
                  S7_1_down, S7_2_down, S7_3_down, 
                  an_pre_down, an_mei_down, an_mic_down, an_bi_down, 
                  ov_1st_down, ov_2nd_down, ov_3rd_down, ov_4th_down,
                  em_dry_down, em_12_down,
                  em_24_down, em_7DAF_down, en_7DAF_down) %>%  
  mutate(location = "3' end")
```


##sample des, library size
```{r}
sample_des <- read_excel("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/sample_des_revision.xlsx")

smRNA_comp <- read_csv("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/code and output/smRNA_comp.csv")

total_siRNA_length <- read_delim("C:/Users/cxli9/Desktop/Li/PBI/PBI299 (Sundar Lab)/smRNA project/CLVS_smRNA_revision/length_profiel_fig2/lengths_summary.bam.siRNAs.sam.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE)
```
```{r}
colnames(total_siRNA_length)[1] <- "library"
total_siRNA_length <- total_siRNA_length %>% 
    mutate(sample_ID = case_when(
    str_detect(library, "sperm.1") ~ "BS1", #1
    str_detect(library, "Sperm2") ~ "BS2",  #2
    str_detect(library, "Sperm4a") ~ "BS4a",  #3
    str_detect(library, "sperm4b") ~ "BS4b",  #4
    str_detect(library, "sperm6a") ~ "BS6a",  #5
    str_detect(library, "sperm6b") ~ "BS6b",  #6
    str_detect(library, "sperm.5b") ~ "SP5b", #7
    str_detect(library, "sperm.5c") ~ "SP5c", #8
    str_detect(library, "Bulk.15") ~ "B15",   #9
    str_detect(library, "Bulk.20") ~ "B20",   #10
    str_detect(library, "Bulk.25") ~ "B25",   #11
    str_detect(library, "seedling4") ~ "SD4", #12
    str_detect(library, "egg.9a") ~ "E9a",    #13
    str_detect(library, "egg.9b") ~ "E9b",    #14
    str_detect(library, "egg.9c") ~ "E9c",    #15
    str_detect(library, "eggcell4") ~ "E4",   #16
    str_detect(library, "eggcell3") ~ "E3",   #17
    str_detect(library, "eggcell2") ~ "E2",   #18
    str_detect(library, "ovary.no.egg.1a") ~ "OV1a",    #19
    str_detect(library, "ovary.no.egg.1b") ~ "OV1b",    #20
    str_detect(library, "Ovarynoegg2") ~ "OV2",         #21
    str_detect(library, "SRR57") ~ "root",              #22
    str_detect(library, "ab_control") ~ "ddm1_con",     #23
    str_detect(library, "ddm1ab") ~ "ddm1",             #24
    str_detect(library, "drm2-T-DNA_control") ~ "drm2_con",  #25 drm2-T-DNA_control.fastq
    str_detect(library, "drm2-T-DNA.fas") ~ "drm2",          #26   
    str_detect(library, "SRR2544786") ~ "em_dry", #27  
    str_detect(library, "SRR5049778") ~ "an_pre",  #28
    str_detect(library, "SRR5049779") ~ "an_mei",  #29
    str_detect(library, "SRR5049780") ~ "an_mic",  #30
    str_detect(library, "SRR5049781") ~ "an_bi",   #31
    str_detect(library, "SRR5049786") ~ "ov_1st",  #32
    str_detect(library, "SRR5049787") ~ "ov_2nd",  #33
    str_detect(library, "SRR5049788") ~ "ov_3rd",  #34
    str_detect(library, "SRR5049789") ~ "ov_4th",  #35
    str_detect(library, "S3_1") ~ "S3_1",          #36
    str_detect(library, "S3_2") ~ "S3_2",          #37
    str_detect(library, "S3_3") ~ "S3_3",          #38
    str_detect(library, "S5_1") ~ "S5_1",          #39
    str_detect(library, "S5_2") ~ "S5_2",          #40
    str_detect(library, "S5_3") ~ "S5_3",          #41
    str_detect(library, "S7_1") ~ "S7_1",          #42
    str_detect(library, "S7_2") ~ "S7_2",          #43
    str_detect(library, "S7_3") ~ "S7_3",          #44
    str_detect(library, "SRR2544787") ~ "em_12h",  #45
    str_detect(library, "SRR2544788") ~ "em_24h",  #46
    str_detect(library, "SRR771501") ~ "em_7DAF",  #47
    str_detect(library, "SRR771500") ~ "en_7DAF"   #48          
  ))

```

```{r}
sample_des_size <- smRNA_comp %>% 
  inner_join(sample_des, by = "sample_ID") %>% 
  select(-library) %>% 
  full_join(total_siRNA_length, by = "sample_ID")

sample_des_size %>% head()
```

#normalize
```{r}
meta <- rbind(upstream, downstream) %>% 
  mutate(location = factor(location, levels = c("5' end", "3' end"))) %>% 
  full_join(sample_des_size, by = "sample_ID") %>% 
  mutate(pro_24 = total.overlap/ `24`) %>% 
  mutate(pro_sm = total.overlap/ `total read`) %>% 
  mutate(pro_si = total.overlap/ `general siRNA`)

meta 
```



#plot
```{r}
axis_line <- data.frame(
  `bin start` = c(-Inf),
  location = c("5' end")
)

axis_line
```

```{r}
meta %>% 
  filter(genotype == "WT") %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot"|
           sample_type == "ovary") %>% 
  ggplot(aes(x = bin.start, y = pro_24 * 1000)) +
  facet_grid(.~location, scales = "free_x")+
  geom_vline(data = axis_line, aes(xintercept = bin.start), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "ribbon",  fun.data = "mean_cl_boot", aes(fill = sample_type), alpha = 0.3) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type, color = sample_type), size = 1.25, alpha = 0.8) +
  scale_x_continuous(breaks = c(-3000,-2000,-1000, 0, 1000, 2000, 2900),
                     labels = NULL)+
  scale_color_manual(values = c("orangered3", "tomato1", "seagreen", "dodgerblue2"),
                    limits = c("ovary", "egg", "seedling shoot", "sperm")) +
  scale_fill_manual(values = c("orangered3", "tomato1", "seagreen", "dodgerblue2"),
                    limits = c("ovary", "egg", "seedling shoot", "sperm")) +
  guides(fill = F) + 
  labs(color = NULL,
       x = NULL, 
       y = "normalized coverage") + 
  theme_minimal()+
  theme(legend.position = c(0.78, 0.8)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black"))

ggsave(filename = "24_meta_gamate.svg", height = 4, width = 4.7)
ggsave(filename = "24_meta_gamate.png", height = 4, width = 4.7)
```
```{r}
#"female"
meta %>% 
  filter(genotype == "WT") %>% 
  filter(sample_type == "egg" |
           str_detect(sample_ID, "ov_") |
           #sample_type == "ovary"|
           sample_type == "seedling shoot" 
           ) %>%  
  mutate(sample_type2 = case_when(
    str_detect(sample_ID, "ov_1st") ~ "pre-meiotic ovary",
    str_detect(sample_ID, "ov_2nd") ~ "meiotic ovary",
    str_detect(sample_ID, "ov_3rd") ~ "functional megaspore\novary",
    str_detect(sample_ID, "ov_4th") ~ "8-nuclei\nembryo sac ovary",
    str_detect(sample_ID, "ov_") == F ~ sample_type,
  )) %>%
  ggplot(aes(x = bin.start, y = pro_24 * 1000)) +
  facet_grid(.~location, scales = "free_x")+
  geom_vline(data = axis_line, aes(xintercept = bin.start), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "ribbon",  fun.data = "mean_cl_boot", aes(fill = sample_type), alpha = 0.3) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type2, color = sample_type2), size = 1.25, alpha = 0.8) +
  scale_x_continuous(breaks = c(-3000,-2000,-1000, 0, 1000, 2000, 2900),
                     labels = NULL)+
  scale_color_manual(values = c("seagreen", 
                               # "orangered3",
                                "orchid1", "brown2", "brown3", "brown4", "tomato1"),
                     limits = c( "seedling shoot", 
                                 #"ovary", 
    "pre-meiotic ovary", "meiotic ovary", 
    "functional megaspore\novary", "8-nuclei\nembryo sac ovary", 
    "egg")) +
  scale_fill_manual(values = c("seagreen", 
                               # "orangered3",
                                "orchid1", "brown2", "brown3", "brown4", "tomato1"),
                     limits = c( "seedling shoot", 
                                 #"ovary", 
    "pre-meiotic ovary", "meiotic ovary", 
    "functional megaspore\novary", "8-nuclei\nembryo sac ovary", 
    "egg")) +
  guides(fill = F) +
  labs(color = NULL,
       x = NULL, 
       y = "normalized coverage") + 
  theme_minimal()+
  theme(legend.position = c(0.74, 0.77)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(legend.text=element_text(size = 13)) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black"))

ggsave(filename = "24_meta_female.svg", height = 4, width = 5)
ggsave(filename = "24_meta_female.png", height = 4, width = 5)
```
```{r}
#"male"
meta %>% 
  filter(genotype == "WT") %>% 
  filter(sample_type == "sperm" |
           str_detect(sample_type, "anther") |
           sample_type == "seedling shoot"
           ) %>% 
  mutate(sample_type2 = case_when(
    sample_type == "bicellular pollen anther" ~ "bicellular pollen\nanther",
    sample_type == "meiosis anther" ~ "meiotic anther",
    sample_type != "bicellular pollen anther" &
      sample_type != "meiosis anther" ~ sample_type
  )) %>% 
  ggplot(aes(x = bin.start, y = pro_24 * 1000)) +
  facet_grid(.~location, scales = "free_x")+
  geom_vline(data = axis_line, aes(xintercept = bin.start), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "ribbon",  fun.data = "mean_cl_boot", aes(fill = sample_type), alpha = 0.3) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type2, color = sample_type2), size = 1.25, alpha = 0.8) +
  scale_x_continuous(breaks = c(-3000,-2000,-1000, 0, 1000, 2000, 2900),
                     labels = NULL)+
  scale_color_manual(values = c("seagreen", 
                               "tan", "tan2", "tan3", "tan4", "dodgerblue2"),
                     limits = c("seedling shoot",
    "pre-meiotic anther",
    "meiotic anther",
    "microspore anther",
    "bicellular pollen\nanther",
    "sperm")) +
  scale_fill_manual(values = c("seagreen", 
                               "tan", "tan2", "tan3", "tan4", "dodgerblue2"),
                     limits = c("seedling shoot",
    "pre-meiotic anther",
    "meiotic anther",
    "microspore anther",
    "bicellular pollen\nanther",
    "sperm")) +
  guides(fill = F) +
  labs(color = NULL,
       x = NULL, 
       y = "normalized coverage") + 
  theme_minimal()+
  theme(legend.position = c(0.74, 0.77)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(legend.text=element_text(size = 13)) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black"))

ggsave(filename = "24_meta_male.svg", height = 4, width = 5)
ggsave(filename = "24_meta_male.png", height = 4, width = 5)
```
```{r}
#spikelet
meta %>% 
  filter(genotype == "WT") %>% 
  filter(sample_type == "sperm" |
           str_detect(sample_type, "spike") |
           sample_type == "seedling shoot"
           ) %>%  
   mutate(sample_type2 = case_when(
    str_detect(sample_ID, "S3") ~ "spikelets\nanther stage 3", 
    str_detect(sample_ID, "S5") ~ "spikelets\nanther stage 5",
    str_detect(sample_ID, "S7") ~ "spikelets\nanther stage 7",
    str_detect(sample_type, "spike") == F ~ sample_type
  )) %>% 
  ggplot(aes(x = bin.start, y = pro_24 * 1000)) +
  facet_grid(.~location, scales = "free_x")+
  geom_vline(data = axis_line, aes(xintercept = bin.start), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "ribbon",  fun.data = "mean_cl_boot", aes(fill = sample_type2), alpha = 0.3) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type2, color = sample_type2), size = 1.25, alpha = 0.8) +
  scale_x_continuous(breaks = c(-3000,-2000,-1000, 0, 1000, 2000, 2900),
                     labels = NULL)+
  scale_color_manual(values = c("seagreen", 
                                "turquoise1", "turquoise3", "turquoise4",
                               "dodgerblue2"),
                     limits = c("seedling shoot",
    "spikelets\nanther stage 3",
    "spikelets\nanther stage 5",
    "spikelets\nanther stage 7",
    "sperm")) +
  scale_fill_manual(values = c("seagreen", 
                                "turquoise1", "turquoise3", "turquoise4",
                               "dodgerblue2"),
                     limits = c("seedling shoot",
    "spikelets\nanther stage 3",
    "spikelets\nanther stage 5",
    "spikelets\nanther stage 7",
    "sperm")) +
  labs(color = NULL,
       x = NULL, 
       y = "normalized coverage") + 
  guides(fill = F) +
  theme_minimal()+
  theme(legend.position = c(0.72, 0.8)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(legend.text=element_text(size = 13)) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black"))

ggsave(filename = "24_meta_spikelet.svg", height = 4, width = 5)
ggsave(filename = "24_meta_spikelet.png", height = 4, width = 5)
```
```{r}
meta %>% 
  filter(genotype == "WT") %>% 
  filter(#sample_type == "sperm" |
           str_detect(sample_type, "spike") 
           #sample_type == "seedling shoot"
           ) %>%  
   mutate(sample_type2 = case_when(
    str_detect(sample_ID, "S3") ~ "spikelets\nanther stage 3", 
    str_detect(sample_ID, "S5") ~ "spikelets\nanther stage 5",
    str_detect(sample_ID, "S7") ~ "spikelets\nanther stage 7",
    str_detect(sample_type, "spike") == F ~ sample_type
  )) %>% 
  ggplot(aes(x = bin.start, y = pro_24 * 1000)) +
  facet_grid(.~location, scales = "free_x")+
  geom_vline(data = axis_line, aes(xintercept = bin.start), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  #stat_summary(geom = "line", fun.y = mean, aes(group = sample_type2, color = sample_type2), size = 1.25, alpha = 0.8) +
  geom_line(aes(group = sample_ID, color = sample_type2), size = 1.25, alpha = 0.8) +
  stat_summary(geom = "errorbar",  fun.data = "mean_cl_boot", aes(group = sample_type2), alpha = 0.8) +
  scale_x_continuous(breaks = c(-3000,-2000,-1000, 0, 1000, 2000, 2900),
                     labels = NULL)+
  scale_color_manual(values = c("turquoise1", "turquoise3", "turquoise4"),
                     limits = c("spikelets\nanther stage 3",
    "spikelets\nanther stage 5",
    "spikelets\nanther stage 7")) +
  labs(color = NULL,
       x = NULL, 
       y = "normalized coverage") + 
  theme_minimal()+
  theme(legend.position = c(0.72, 0.8)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(legend.text=element_text(size = 13)) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black"))
```

```{r}
meta %>% 
  filter(genotype == "WT") %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot" |
           sample_type == "mature embryo"|
           sample_type == "12h imbibed embryo"|
           sample_type == "24h imbibed embryo") %>% 
  ggplot(aes(x = bin.start, y = pro_24 * 1000)) +
  facet_grid(.~location, scales = "free_x")+
  geom_vline(data = axis_line, aes(xintercept = bin.start), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "ribbon",  fun.data = "mean_cl_boot", aes(fill = sample_type), alpha = 0.3) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type, color = sample_type), size = 1.25, alpha = 0.8) +
  scale_x_continuous(breaks = c(-3000,-2000,-1000, 0, 1000, 2000, 2900),
                     labels = NULL)+
  scale_color_manual(values = c("tomato1", "seagreen", "dodgerblue2", "magenta4", "yellow3", 
                                "yellow4"),
                     limits = c("egg", "seedling shoot", "sperm",
                                "mature embryo", 
                                "12h imbibed embryo", 
                                "24h imbibed embryo")) +
  scale_fill_manual(values = c("tomato1", "seagreen", "dodgerblue2", "magenta4", "yellow3", 
                                "yellow4"),
                     limits = c("egg", "seedling shoot", "sperm",
                                "mature embryo", 
                                "12h imbibed embryo", 
                                "24h imbibed embryo")) +
  guides(fill = F) +
  labs(color = NULL,
       x = NULL, 
       y = "normalized coverage") + 
  theme_minimal()+
  theme(legend.position = c(0.72, 0.8)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black"))

ggsave(filename = "24_meta_gamate_em.svg", height = 4, width = 5)
ggsave(filename = "24_meta_gamate_em.png", height = 4, width = 5)
```
```{r}
meta %>% 
  filter(genotype == "WT") %>% 
  filter(sample_type == "egg"|
           sample_type == "sperm"|
           sample_type == "seedling shoot" |
           sample_type == "mature embryo"|
           sample_type == "embryo 7-8 DAF"|
           sample_type == "endosperm 7-8 DAF") %>% 
  ggplot(aes(x = bin.start, y = pro_24 * 1000)) +
  facet_grid(.~location, scales = "free_x")+
  geom_vline(data = axis_line, aes(xintercept = bin.start), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "ribbon",  fun.data = "mean_cl_boot", aes(fill = sample_type), alpha = 0.3) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type, color = sample_type), size = 1.25, alpha = 0.8) +
  scale_x_continuous(breaks = c(-3000,-2000,-1000, 0, 1000, 2000, 2900),
                     labels = NULL)+
  scale_color_manual(values = c("tomato1", "seagreen", "dodgerblue2", "magenta4", "magenta", 
                                "lightgoldenrod4"),
                     limits = c("egg", "seedling shoot", "sperm",
                                "mature embryo", 
                                "embryo 7-8 DAF", 
                                "endosperm 7-8 DAF")) +
  scale_fill_manual(values = c("tomato1", "seagreen", "dodgerblue2", "magenta4", "magenta", 
                                "lightgoldenrod4"),
                     limits = c("egg", "seedling shoot", "sperm",
                                "mature embryo", 
                                "embryo 7-8 DAF", 
                                "endosperm 7-8 DAF")) +
  labs(color = NULL,
       x = NULL, 
       y = "normalized coverage") + 
  guides(fill = F) + 
  theme_minimal()+
  theme(legend.position = c(0.72, 0.8)) +
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black"))

ggsave(filename = "24_meta_gamate_em_en.svg", height = 4, width = 5)
ggsave(filename = "24_meta_gamate_em_en.png", height = 4, width = 5)
```


```{r}
meta %>% 
  filter(genotype == "WT") %>% 
  filter(sample_type == "seedling shoot") %>% 
  ggplot(aes(x = bin.start, y = pro_24 * 1000)) +
  facet_grid(.~location, scales = "free_x")+
  geom_vline(data = axis_line, aes(xintercept = bin.start), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  stat_summary(geom = "line", fun.y = mean, aes(group = sample_type, color = sample_type), size = 1.25, alpha = 0.8) +
  #stat_summary(geom = "linerange",  fun.data = "mean_cl_boot", aes(color = sample_type)) +
  scale_x_continuous(breaks = c(-3000,-2000,-1000, 0, 1000, 2000, 2900),
                     labels = NULL)+
  scale_color_manual(values = "seagreen") +
  labs(color = NULL,
       x = NULL, 
       y = "normalized coverage") + 
  theme_minimal()+
  theme(legend.position = "none") +
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black"))

ggsave(filename = "24_meta_seedling.svg", height = 4, width = 5)
ggsave(filename = "24_meta_seedling.png", height = 4, width = 5)
```

```{r}
meta_m <- meta %>% 
  filter(sample_type == "leaf") %>% 
  mutate(rows = case_when(
    str_detect(sample_ID, "ddm") ~ "ddm1ab leaf",
    str_detect(sample_ID, "drm") ~ "drm2 leaf"
  )) 
```

```{r}
meta_m %>% 
  filter(sample_type == "leaf") %>% 
  ggplot(aes(x = bin.start, y = pro_24 * 1000)) +
  facet_grid(rows ~ location, scales = "free_x")+
  geom_vline(data = axis_line, aes(xintercept = bin.start), size = 1.5) +
  geom_hline(yintercept = -Inf, size = 1.5) +
  geom_line(aes(color = genotype), size = 1, alpha = 0.7) +
  scale_x_continuous(breaks = c(-3000,-2000,-1000, 0, 1000, 2000, 2900),
                     labels = NULL)+
  scale_color_manual(values = c("red", "blue", "grey40")) +
  labs(color = NULL,
       x = NULL, 
       y = "normalized coverage") + 
  theme_minimal()+
  theme(legend.position = c(0.85, 0.4), strip.text.y = element_blank())+
  theme(text = element_text(size = 18, face="bold")) +
  theme(axis.text.x=element_text(colour="black")) +
  theme(axis.text.y=element_text(colour="black")) 

ggsave(filename = "ddm1 and drm2 24_meta_WT.svg", height = 4, width = 5)
ggsave(filename = "ddm1 and drm2 24_meta_WT.png", height = 4, width = 5)
```



Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
